test/transcoding/annotate_attribute.ll

```diff
@@ -205,6 +211,30 @@ entry:
   ret i32 0
 }
 
+; Function Attrs: mustprogress norecurse
+define weak_odr dso_local spir_kernel void @_ZTSZ11TestKernelAvE4MyIP(i32 addrspace(1)* noundef align 4 %0) local_unnamed_addr #5 !kernel_arg_buffer_location !15 !sycl_kernel_omit_args !16 {
+  %2 = alloca %struct.MyIP, align 8
+  %3 = bitcast %struct.MyIP* %2 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 8, i8* nonnull %3) #4
+  %4 = addrspacecast i8* %3 to i8 addrspace(4)*
+  %5 = call i8 addrspace(4)* @llvm.ptr.annotation.p4i8(i8 addrspace(4)* %4, i8* getelementptr inbounds ([4 x i8], [4 x i8]* @.str.11, i64 0, i64 0), i8* getelementptr inbounds ([9 x i8], [9 x i8]* @.str.1.12, i64 0, i64 0), i32 13, i8* bitcast ({ i32, i32, i32 }* @.args to i8*))
+  ; CHECK-LLVM: %[[ALLOCA:.*]] = alloca %struct.MyIP, align 8
+  ; CHECK-LLVM: %[[BITCAST:.*]] = bitcast ptr %[[ALLOCA]] to ptr
+  ; CHECK-LLVM: %[[ADDRSPACECAST:.*]] = addrspacecast ptr %[[BITCAST]] to ptr addrspace(4)
+  ; CHECK-LLVM: %[[UNADDRSPACECAST:.*]] = addrspacecast ptr addrspace(4) %[[ADDRSPACECAST]] to ptr
+  ; CHECK-LLVM: call void @llvm.var.annotation(ptr %[[UNADDRSPACECAST]], ptr [[STR12]], ptr undef, i32 undef, ptr undef)
```

Comment:
In the original LLVM IR here is `llvm.ptr.annotation` and not `llvm.var.annotation`.

I believe the issue is that to access 0th member of the structure clang doesn't generate GEP(0, 0) and clang is correct in doing this, because GEP(0, 0) is quite useless instruction. So the algorithm in the reader should be adjusted - once we reach alloca it should check whether it's alloca for struct or not.

Reply:
Ah, I've run across this 0th struct member problem in other transforms before. I've updated the loop to only set `isStaticMemoryAttribute` if the base alloca is not a struct type.